src/Selection.js

```diff
@@ -243,17 +244,46 @@ class Selection {
       return this.emit('reset')
     }
 
-    if(click && inRoot)
-      return this.emit('click', {
+    if(click && inRoot) {
+      return this._handleClickEvent(e);
+    }
+
+    // User drag-clicked in the Selectable area
+    if(!click)
+      return this.emit('select', bounds)
+  }
+
+  _handleClickEvent(e) {
+    const { pageX, pageY, clientX, clientY } = getEventCoordinates(e);
+    const now = new Date().getTime();
+
+    if (
+        this._lastClickData &&
+        !this._lastClickData.isDblClick &&
```

Comment:
why add this `isDblClick` instead of setting `_lastClickData` to `null` after a double click

Reply:
I don't see much difference between checking for a flag and checking for null value. `_lastClickData = null` seems obscure for me

Reply:
I here you, though the opposite strikes me as more obscure, at the moment we are tracking some metadata that isn't used which isn't clear. I think i'd be clearer to clear out the last click data if there is no last click data :)

Reply:
Fixed. Clear _lastClickData on double click event